common/lib/xmodule/xmodule/license.py

```diff
@@ -0,0 +1,138 @@
+"""
+Different types of licenses for content
+
+This file contains a base license class as well as a some useful specific license classes, namely:
+    - ARRLicense (All Rights Reserved License)
+    - CCLicense (Creative Commons License)
+
+The classes provide utility funcions for dealing with licensing, such as getting an image representation
+of a license, a url to a page describing the specifics of the license, converting licenses to and from json
+and storing some vital information about licenses, in particular the version.
+"""
+from django.utils.translation import ugettext as _
+
+from xblock.fields import JSONField
+
+
+class License(JSONField):
+    """
+    Base License class
+    """
+
+    _default = None
+    MUTABLE = False
+
+    def __init__(self, kind=None, version=None, *args, **kwargs):
+        self.kind = kind
+        self.version = version
+        super(License, self).__init__(*args, **kwargs)
+
+    @property
+    def html(self):
+        """
+        Return a piece of html that describes the license
+
+        This method should be overridden in child classes to provide the desired html.
+        """
+        return u"<p>" + _("This resource is not licensed.") + u"</p>"
+
+    def to_json(self, value):
+        """
+        Return a JSON representation of the license
+        """
+        if value is None:
+            return {"kind": "ARR", "version": "1.0"}
+        elif isinstance(value, License):
+            return {"kind": value.kind, "version": value.version}
+        elif isinstance(value, dict) and 'kind' in value and 'version' in value:
+            return {"kind": value['kind'], "version": value['version']}
+        else:
+            raise TypeError("Cannot convert {!r} to json".format(value))
+
+    def from_json(self, field):
+        """
+        Construct a new license object from a valid JSON representation
+        """
+        return parse_license(field)
+
+    enforce_type = from_json
+
+
+class ARRLicense(License):
+    """
+    License class for an 'All rights reserved' license
+    """
+
+    def __init__(self, kind="ARR", version=None, *args, **kwargs):
+        super(ARRLicense, self).__init__(kind, version, *args, **kwargs)
+
+    @property
+    def html(self):
+        """
+        Return a piece of html that descripts the license
+        """
+        phrase = _("All rights reserved")
+        return "&copy;<span class='license-text'>{phrase}</span>".format(
```

Comment:
Yes, this method will need to be removed. Is not executed by any method call at the moment.